tracing-fmt/src/format.rs

```diff
@@ -44,6 +44,24 @@ impl<N> FormatEvent<N>
     }
 }
 
+pub trait FormatFields<'writer> {
+    fn format_fields<R: RecordFields>(
+        &self,
+        writer: &'writer mut dyn fmt::Write,
```

Comment:
Is there a reason to be generic over the fields but not over the writer?

Reply:
Yes, the writer is the `MakeVisitor` trait's target type, which the trait is generic over. This means that we would have to make the `FormatFields` _trait_ (rather than the function) generic over a writer, and thus also make the `FmtSubscriber` itself generic over a writer. This won't work since we will need to pass multiple writer types in (sometimes a `String`, and sometimes an io stream), so we need to use a trait object to erase the concrete type. On the other hand, the `RecordFields` is only a parameter of the _method_, so implementations of `FormatFields` can't implement the trait only for specific implementors of `RecordFields`. I thought it was better to avoid the overhead of a trait object or that since it's not strictly required.

Reply:
We could make the method generic over both a writer and a `RecordFields`, but the writer will _eventually_ become a trait object if we use it to create a visitor. That might, actually, be better, as implementations that _don't_ use the blanket impl for `MakeVisitor` could avoid a trait object...